ynput/ayon-maya
# Support to create face sets for materials in Extract Alembic

### Changelog Descriptions
This PR is to add support for add materials per face sets for in Extract Alembic when the "write face sets" option is enabled. It also adds the optional validator to check on the write face sets options. Users can turn on the validator when they want to export the alembic to Unreal Engine to make sure the write face sets options switched on.

If the `write face sets` option is enabled with this PR, it will assign the material slots per each mesh while the mesh is merged.
![image](https://github.com/user-attachments/assets/fb522a71-8275-48e7-8d8f-93cbdb994d5c)
If the `write face sets` option is disabled, it will result one shader in merged mesh
![image](https://github.com/user-attachments/assets/423c9d99-5b96-43dd-8cdd-2747c9e5c4e6)

### Additional Info
It is an extension of fix for this issue lasted in ayon-unreal
https://github.com/ynput/ayon-unreal/issues/21#issuecomment-2242540494
The related enhancement of the PR created for ayon_unreal: https://github.com/ynput/ayon-unreal/pull/25
### Testing Notes
1. Launch Maya
2. Create Model/Point Cache
3. Make sure the write face sets turned on
4. Publish
5. Loaded the model in Unreal
6. There should be multiple material slots if there are materials/shaders assigned.
